File: types/tourist-spot.ts
// Tourist Spot Types

export interface TouristSpot {
  id: number;
  name: string;
  description: string;
  imageUrl: string;
  rating: number;
  location: string;
  coordinates?: string;
  address: string;
  openingHours?: string;
  entryFee?: string;
  highlights: string[];
  travelTime?: string;
  isActive: boolean;
  viewCount: number;
  createdAt: string;
  updatedAt: string;
}

export interface CreateTouristSpotRequest {
  name: string;
  description: string;
  imageUrl: string;
  rating?: number; // defaults to 5.0
  location: string;
  coordinates?: string;
  address: string;
  openingHours?: string;
  entryFee?: string;
  highlights?: string[];
  travelTime?: string;
  isActive?: boolean; // defaults to true
}

export interface CreateTouristSpotWithImageRequest {
  name: string;
  description: string;
  rating?: number; // defaults to 5.0
  location: string;
  coordinates?: string;
  address: string;
  openingHours?: string;
  entryFee?: string;
  highlights?: string[];
  travelTime?: string;
  isActive?: boolean; // defaults to true
  image?: File; // Optional image file
}

export interface UpdateTouristSpotRequest {
  name: string;
  description: string;
  imageUrl: string;
  rating?: number; // defaults to 5.0
  location: string;
  coordinates?: string;
  address: string;
  openingHours?: string;
  entryFee?: string;
  highlights?: string[];
  travelTime?: string;
  isActive?: boolean; // defaults to true
}

export interface UpdateTouristSpotWithImageRequest {
  name: string;
  description: string;
  rating?: number; // defaults to 5.0
  location: string;
  coordinates?: string;
  address: string;
  openingHours?: string;
  entryFee?: string;
  highlights?: string[];
  travelTime?: string;
  isActive?: boolean; // defaults to true
  image?: File; // Optional image file
}

export interface TouristSpotListResponse {
  success: boolean;
  data: TouristSpot[];
  message: string;
}

export interface TouristSpotResponse {
  success: boolean;
  data: TouristSpot;
  message: string;
}

export interface TouristSpotFilters {
  page?: number;
  pageSize?: number;
  isActive?: boolean;
} 

File: Alisto.Api/Models/TouristSpot.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json;

namespace Alisto.Api.Models
{
    public class TouristSpot
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(200)]
        public string Name { get; set; }

        [Required]
        public string Description { get; set; }

        [MaxLength(500)]
        public string ImageUrl { get; set; } = string.Empty;

        [Column(TypeName = "decimal(3,2)")]
        public decimal Rating { get; set; } = 5.0m;

        [Required]
        [MaxLength(200)]
        public string Location { get; set; }

        [MaxLength(100)]
        public string? Coordinates { get; set; }

        [Required]
        [MaxLength(500)]
        public string Address { get; set; }

        [MaxLength(100)]
        public string? OpeningHours { get; set; }

        [MaxLength(100)]
        public string? EntryFee { get; set; }

        [Column(TypeName = "json")]
        public string Highlights { get; set; } = "[]";

        [MaxLength(100)]
        public string? TravelTime { get; set; }

        public bool IsActive { get; set; } = true;

        public int ViewCount { get; set; } = 0;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Helper methods for Highlights
        [NotMapped]
        public List<string> HighlightsList
        {
            get => JsonSerializer.Deserialize<List<string>>(Highlights) ?? new List<string>();
            set => Highlights = JsonSerializer.Serialize(value);
        }
    }
}